protocols/imap/src/main/java/org/apache/james/imap/processor/AbstractMailboxProcessor.java

```diff
@@ -83,9 +88,12 @@ protected final void doProcess(M acceptableMessage, Responder responder, ImapSes
     }
 
     protected final void process(M message, Responder responder, ImapSession session) {
-        final ImapCommand command = message.getCommand();
-        final String tag = message.getTag();
+        ImapCommand command = message.getCommand();
+        String tag = message.getTag();
+
+        TimeMetric timeMetric = metricFactory.timer(command.getName());
         doProcess(message, command, tag, responder, session);
+        timeMetric.elapseTimeInMs();
```

Comment:
Then, you are linked to the binded implementation -> log.
The idea is just to have a timer metric and just start & stop method, then you don't take care of what is doing the implementation with this data.